### PR TITLE
Morgue tray now updates properly

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Morgue tray now updates properly when you re-enter/exit your body.
Fixes: #22876

Currently a draft PR so I can fix some remaining issues

- [x] Update morgue tray state when you DNR
- [x] Use signals to update the morgue tray instead of checking whether we are in a morgue

## Why It's Good For The Game
Morgue trays should accurately represent whether the corpse inside is currently revivable, or could viably be revived in the future.

## Testing
Enter a morgue tray and die.
Morgue tray light is now green.
Exit my corpse.
Morgue tray light is now purple.

## Changelog
:cl:
fix: Morgue tray now updates properly
/:cl:

### CODE REVIEW

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -48,34 +48,34 @@
 
 	if(connected)
 		status = EXTENDED_TRAY
-		return
 
-	if(!length(contents))
+	else if(!length(contents))
```
Comment (Contrabang) [inline code/game/objects/structures/morgue.dm:52]: Please return these to guard-clauses, and just put an update icon in them

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -48,34 +48,34 @@
...
-		M = locate() in B
+		if(!M)
+			M = locate() in B
 
-	if(!M)
-		status = NOT_BODY
-		return
+		if(!M)
+			status = NOT_BODY
+			return update_icon(UPDATE_OVERLAYS)
```
Comment (Contrabang) [inline code/game/objects/structures/morgue.dm:64]:
```suggestion
			update_icon(UPDATE_OVERLAYS)
			return
```

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -48,34 +48,34 @@
...
-		return
 
-	if(!length(contents))
+	else if(!length(contents))
 		status = EMPTY_MORGUE
-		return
 
-	var/mob/living/M = locate() in contents
-	var/obj/structure/closet/body_bag/B = locate() in contents
+	else
```
Comment (Contrabang) [inline code/game/objects/structures/morgue.dm:55]:
```suggestion
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -59,6 +59,7 @@
 	if(.)
 		if(ranged_ability && prev_client)
 			ranged_ability.remove_mousepointer(prev_client)
+	SEND_SIGNAL(src, COMSIG_LIVING_UPDATE_REVIVABILITY)
```
Comment (Contrabang) [inline code/modules/mob/living/living.dm:62]: Could you make this signal name more generalized? Something like `COMSIG_LIVING_GHOSTIZED`

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/__DEFINES/dcs/signals.dm [hunk 5]
```diff
@@ -451,6 +451,8 @@
 //sent from living mobs when they are ahealed
 #define COMSIG_LIVING_AHEAL "living_aheal"
 
+#define COMSIG_LIVING_UPDATE_REVIVABILITY "revivablity"
```
Comment (Contrabang) [inline code/__DEFINES/dcs/signals.dm:454]: Also, this needs a comment, something like this
```suggestion
/// sent from the end of mob/living/ghostize: ()
#define COMSIG_LIVING_UPDATE_REVIVABILITY "revivablity"
```

File: code/modules/mob/dead/observer/observer_base.dm [hunk 6]
```diff
@@ -301,11 +294,7 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	mind.current.key = key
 
-	var/obj/structure/morgue/Morgue = locate() in mind.current.loc
-	if(istype(mind.current.loc,/obj/structure/morgue))
-		Morgue = mind.current.loc
-	if(Morgue)
-		Morgue.update_state()
+	SEND_SIGNAL(mind.current, COMSIG_LIVING_REENTERED_BODY)
```
Comment (Sirryan2002) [inline code/modules/mob/dead/observer/observer_base.dm:297]: we love better code :heart:

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**